### PR TITLE
Tokio tls http2 rewrite

### PR DESCRIPTION
**Pull request by technetos**
This PR removes hyper and implements http2 over tls using tokio.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,55 +11,48 @@ $ git clone https://github.com/technetos/catalyst.git
...
 
-Lets go ahead and run the `simple_routes` example now:
+To run the `simple routes` example:
 ```
-$ cargo r --example simple_routes
-> Server running at: 127.0.0.1:8000
+$ cargo r --release --example simple_routes
+Listening on: 127.0.0.1:3000
 ```
+The simple routes example has the following routes
```
Comment (rupsis) [inline README.md]:
```suggestion
The `simple_routes` example has the following routes
```

File: src/router.rs [hunk 2]
```diff
@@ -1,90 +1,75 @@
...
         }
     }
 
-    /// Returns a reference to the HTTP method for this route.  
-    fn method(&self) -> &Method {
-        &self.method
+    pub fn add(&mut self, uri: &str, method: &str, fn_ptr: route_fn) {
+        self.routes.insert(
+            (String::from(uri), String::from(method)),
+            fn_ptr as route_fn,
```
Comment (rupsis) [inline src/router.rs:27]: 👌

**Review by rupsis — DISMISSED** [review]

File: README.md [hunk 3]
```diff
@@ -11,55 +11,48 @@ $ git clone https://github.com/technetos/catalyst.git
...
 ## Routes
 
 Functions can be used as routes in Catalyst as long as they match the signature
-`(&Request) -> Response`
+`(Request<Json>) -> Response`
 
 ## Basic Usage
 
 ```rust
-use catalyst::{router::Router, server::Server};
```
Comment (rupsis) [inline README.md:41]: These imports are still needed for basic usage, correct?

File: src/router.rs [hunk 4]
```diff
@@ -1,90 +1,75 @@
...
-where
-    for<'r> F: Fn(&'r Request<Body>) -> Response<Body>,
-{
-    fn respond_to<'r>(&self, req: &'r Request<Body>) -> Response<Body> {
-        self(req)
-    }
-}
+pub type route_fn = fn(Request<Json>) -> Response;
 
-/// # Route
```
Comment (rupsis) [inline src/router.rs:22]: Could you keep the docs? or do you anticipate some major changes in the future?

**Review by rupsis — APPROVED** [review]
Changes look good. Just a few questions about some documentation.

File: src/router.rs (hunk 4 above)
Comment (technetos) [reply src/router.rs:22]: Its going to be overhauled

**Pull request merged**